### PR TITLE
Enhancement/64 test generate json file

### LINKED ISSUE #64: Testing | Add unit tests for "ConfigFile.generateJSONFile" method
Create units test to verify that the `ConfigFile.generateJSONFile` method works as expected:

- [ ] Clear the Local Store, populate it using `ConfigFile.parseJSONFile` (and some sample configuration file), then call `ConfigFile.generateJSONFile` and then verify than the result is equal to the sample configuration file.
- [ ] Clear the Local Store, populate it using `ConfigFile.parseJSONFile` (and some sample configuration file), make some changes on the Local Store, then call `ConfigFile.generateJSONFile` and then verify than the result is equal to the sample configuration file plus the changes previously done.

Blocked by #58, related to #57

### PR DESCRIPTION
**Pull request by Saidmurod1996**
Fixes #64

### CODE REVIEW

**Comment by tim-kodirov** [comment]
Tests are running successfully

File: tests/unit/configFile.spec.js [hunk 1]
```diff
@@ -6,6 +8,7 @@ const ConfigFile = require("../../src/js/shared/config_file");
 const defaultSettings = require("../resources/defaultSettings.json");
 const encryptedSettings = require("../resources/encryptedSettings.json");
 
+const generateJSONFile = jest.spyOn(ConfigFile.default, "generateJSONFile");
```
Comment (jfmdev) [inline tests/unit/configFile.spec.js:11]: No need to add an spy to `generateJSONFile` if the spy-related methods will never be called

File: tests/unit/configFile.spec.js [hunk 2]
```diff
@@ -15,69 +18,96 @@ describe("configFile.js", () => {
...
-    const result = await updateNow();
-    var lastConfigData = await LocalStore.getOne(StoreKey.LAST_CONFIG_DATA);
-    expect(lastConfigData.errorMsg).not.toBe(null);
-    expect(result).toBe(false);
+  describe("generateJSONFile function", () => {
+    it("generateJSONFile result should be equal to simple configuration file", async () => {
+      await LocalStore.clear();
+      await parseJSONFile(defaultSettings, true);
+
+      const result = await generateJSONFile();
```
Comment (jfmdev) [inline tests/unit/configFile.spec.js:94]: If spy-related methods won't be used, here you can just use `ConfigFile.default.parseJSONFile` and `ConfigFile.default.generateJSONFile`

File: tests/unit/configFile.spec.js [hunk 3]
```diff
@@ -15,69 +18,96 @@ describe("configFile.js", () => {
...
+    });
+
+    it("generateJSONFile result should be equal to changed configuration file", async () => {
+      await LocalStore.clear();
+      await parseJSONFile(defaultSettings, true);
+
+      // Change data in local storage.
+      var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      settings.configEncrypted = true;
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
```
Comment (jfmdev) [inline tests/unit/configFile.spec.js:105]: Please change more things... 

* Change two or three values from settings.
* Change the name of one group.
* Remove a search provider, add a new search provider and modify the name of an existing search provider.
* Change a special provider (modify two values from Config and on Queries remove a value and update an existing one).

File: tests/unit/configFile.spec.js [hunk 4]
```diff
@@ -15,69 +18,96 @@ describe("configFile.js", () => {
...
+
+      const changedSettings = _.omit(defaultSettings, "update");
+      changedSettings.config[0][2] = true;
+      const result = await generateJSONFile();
```
Comment (jfmdev) [inline tests/unit/configFile.spec.js:109]: Same as above: you can just use `ConfigFile.default.parseJSONFile` and `ConfigFile.default.generateJSONFile`

**Review by jfmdev — CHANGES_REQUESTED** [review]

**Review by jfmdev — APPROVED** [review]

**Pull request merged**